src/libcore/option.rs

```diff
@@ -178,6 +178,7 @@ impl<T> Option<T> {
     /// ```
     ///
     /// [`Some`]: #variant.Some
+    #[must_use]
```

Comment:
Might be a good idea to add a message here explaining that this function has no side-effects, or so?

Reply:
Agreed!  Maybe with a pointer to `unwrap` or `expect` if people meant it to be an assertion?

Reply:
Did this ever get resolved?  If not someone should open a pr for it.

Reply:
Nope, there's still no message. Can you open an issue or a PR?

Reply:
I would but I honestly have no clue what to put for the message.

Reply:
@czipperz Well, inspired by https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect, here's a proposal for the text:

- `Result::is_ok`: if you intended to assert that this matches `Ok(_)`, consider `.unwrap()` instead
- `Result::is_err`: if you intended to assert that this matches `Err(_)`, consider `.unwrap_err()` instead
- `Option::is_some`: if you intended to assert that this matches `Some(_)`, consider `.unwrap()` instead
- `Option::is_none`: if you intended to assert that this matches `None`, consider ```.or_else(|| panic!("called `Option::unwrap()` on a `None` value"))``` instead

(That last one makes me think of https://internals.rust-lang.org/t/adding-option-expect-none/10481?u=scottmcm...)

Reply:
I'll open a PR based on those messages tonight if someone else doesn't do it first.